pilot/pkg/model/push_context.go

```diff
@@ -97,6 +97,9 @@ type PushContext struct {
 	// ServiceAccounts contains a map of hostname and port to service accounts.
 	ServiceAccounts map[host.Name]map[int][]string `json:"-"`
 
+	// ServiceAccountAliases contains a map of hostname and port to service accounts aliases.
+	ServiceAccountAliases map[host.Name]map[int][]string `json:"-"`
```

Comment:
why do you need this separate array? Why not add this into the same array?

Reply:
I was trying to distinguish between the real trust domain and its aliases, but I guess there is no point in doing so. I'll reuse the same array. Thanks.

/cc @liminw